Add option to add specific center or doctor

Dreiundzwanzig opened the pull request:
Add options to:
- add specific centers or doctors

Closes #68, closes #82

danburg committed:
Add option to add additional center or doctor

gitolicious commented on doctoshotgun.py:
```diff
@@ -335,6 +335,17 @@ def find_centers(self, where, motives=None, page=1):
 
+        if additional_centers:
+            for additional_center in additional_centers:
+                splitted = additional_center.split(';')
+                if len(splitted) == 3:
+                    yield {
+                        "name_with_title": splitted[0], #"Corona Impfzentren - Berlin",
+                        "city": splitted[1], #"Berlin",
+                        "url": splitted[2], #"/institut/berlin/ciz-berlin-berlin"
+                    }
```
I prefer the more explicit nature of named capture groups here, though both solutions do work just fine.
```suggestion
                match = re.match(r'(?P<name>.+);(?P<city>.+);(?P<url>.+)', additional_center)
                if match:
                    yield {
                        "name_with_title": match['name'], #"Corona Impfzentren - Berlin",
                        "city":            match['city'], #"Berlin",
                        "url":             match['url'],  #"/institut/berlin/ciz-berlin-berlin"
                    }
```

Dreiundzwanzig committed:
Use named capture groups

Co-authored-by: gitolicious <26963495+gitolicious@users.noreply.github.com>

rbignon commented:
Hello,

Thank you for your patch. However, I think it is not user friendly to ask user providing the name and city, as the url is enough to guess them. In my opinion, the code may be a little refactored to not apply args filters (--center, --center-regex, etc.) on ones provided with --aditional-center.

Another question related to the usecase itself, do you really want to add these centers to the search results, or are you targeting specifically these centers? Perhaps instead of --additional-center, it would be --center-url and in this case, do not do any search at all?

Dreiundzwanzig committed:
for additional center only url is required

Dreiundzwanzig commented:
Hello,

i've added them to the search result, because at the time of writing the code i didn't want to introduce as as little as possible merge conflicts, but thats not neccesary anymore if it ever was. 

My personal usecase was to add additional centers/doctors from which i knew there offering vaccinations, but are not included in the search results and i wanted them to be processed additionally to the search results.

You are right, i've refactored the code and removed requirement for name and city.

gitolicious commented on doctoshotgun.py:
```diff
@@ -653,6 +663,8 @@ def main(self, cli_args=None):
                             action='append', help='exclude centers')
         parser.add_argument('--center-exclude-regex',
                             action='append', help='exclude centers by regex')
+        parser.add_argument('--additional-center', '-ac',
```
Until now, we only had single-letter abbreviations for arguments but it looks like we are running out of usable letters. Still it should be a deliberate decision to extend to two-letter abbreviations. @rbignon what do you think?

gitolicious commented on doctoshotgun.py:
```diff
@@ -617,6 +617,16 @@ def setup_loggers(self, level):
         logging.root.setLevel(level)
         logging.root.addHandler(self.create_default_logger())
 
+    def try_to_book_or_sleep(self, docto, center, vaccine_list, start_date, end_date, only_second, only_third, dry_run=False):
```
To me the method name is a bit misleading. "or sleep" in fact describes the internal logic of a lazy loading `try_to_book || sleep` but from the sound of it I would expect "or sleep" is related to the return value. Maybe someone comes up with a better name?

Dreiundzwanzig replied:
Yeah your right, i guess we could just use `try_to_book`? I mean it's just that and in fact it does not matter if there is a `sleep` inside or not. What do you think?
